### PR TITLE
feat(moveSegmentProviderStop): Implement moveSegmentProviderStop method

### PR DESCRIPTION
**Pull request by moussamew**
implement `moveSegmentProviderStop` method

### CODE REVIEW

File: src/classes/groundplaces.ts [hunk 1]
```diff
@@ -149,16 +151,16 @@ export class GroundPlacesController {
...
+    const cloneGroundPlaces: GroundPlace[] = cloneDeep(this.getGroundPlaces());
 
-    this.storageService.updatePlace(stopGroupGpuid, propertiesToUpdate);
-
-    // const isUpdateValid: boolean = this.checkValidity();
+    try {
+      this.storageService.updatePlace(stopGroupGpuid, propertiesToUpdate, GroundPlaceType.GROUP);
+    } catch (error) {
+      // If there is error in the process, rollback to the previous version of the ground places stored
+      this.storageService.setGroundPlaces(cloneGroundPlaces);
```
Comment (Theo-Le-Donne) [inline src/classes/groundplaces.ts:160]: 👍

File: __tests__/groundplaces/moveSegmentProviderStop.spec.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    try {
+      groundPlacesService.moveSegmentProviderStop(segmentProviderStopId, fromStopGroupGpuid, intoStopGroupGpuid);
+    } catch (error) {
+      thrownError = error;
+    }
+
+    expect(thrownError).toEqual(new Error('The StopGroup with the Gpuid "g|FRstraroet@u0tkr3333" is not found.'));
+  });
+
+  it(`should throw an error if the SegmentProviderStop specified doesn't belong to the current StopCluster parent specified`, () => {
```
Comment (rimiti) [inline __tests__/groundplaces/moveSegmentProviderStop.spec.ts:173]: Style: use single quotes

File: __tests__/groundplaces/moveStopGroup.spec.ts [hunk 3]
```diff
@@ -228,7 +228,7 @@ describe('#moveStopGroup', () => {
 
     expect(thrownError).toEqual(
       new Error(
-        `You can't move the StopGroup with the Gpuid "g|FRststbi__@u0tkxd" because it already exists inside the new StopCluster parent specified.`,
+        `You can't move the StopGroup with the Gpuid "g|FRststbi__@u0tkxd" because the new StopCluster parent is the same as before.`,
```
Comment (rimiti) [inline __tests__/groundplaces/moveStopGroup.spec.ts:231]: 👍 

> juste use single quotes when you don't have any variables.

File: src/classes/groundplaces.ts [hunk 4]
```diff
@@ -149,16 +151,16 @@ export class GroundPlacesController {
...
-    // const copyGroundPlaces: GroundPlaces = this.storageService.cloneGroundPlaces();
+    const cloneGroundPlaces: GroundPlace[] = cloneDeep(this.getGroundPlaces());
 
-    this.storageService.updatePlace(stopGroupGpuid, propertiesToUpdate);
-
-    // const isUpdateValid: boolean = this.checkValidity();
+    try {
+      this.storageService.updatePlace(stopGroupGpuid, propertiesToUpdate, GroundPlaceType.GROUP);
+    } catch (error) {
+      // If there is error in the process, rollback to the previous version of the ground places stored
```
Comment (rimiti) [inline src/classes/groundplaces.ts:159]: > // If there is an error, previous update is reverted

File: src/classes/groundplaces.ts [hunk 5]
```diff
@@ -304,6 +292,74 @@ export class GroundPlacesController {
...
+      );
+    }
+
+    const cloneGroundPlaces: GroundPlace[] = cloneDeep(this.getGroundPlaces());
+
+    try {
+      const currentStopGroupParent: StopGroup = this.storageService.getStopGroupByGpuid(fromStopGroupGpuid);
+      const newStopGroupParent: StopGroup = this.storageService.getStopGroupByGpuid(intoStopGroupGpuid);
+
+      const segmentProviderStopIndex: number = currentStopGroupParent.childs.findIndex(
```
Comment (rimiti) [inline src/classes/groundplaces.ts:322]: It do the job, but you also can use .map() function if you want.

**Review by rimiti — APPROVED** [review]
A bit complicated to understand all of your classes logic. 
Otherwise, LGTM.

**Comment by antonindurieux** [comment]
LGTM

File: __tests__/groundplaces/moveSegmentProviderStop.spec.ts [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+  });
+
+  it('should throw an error if the current StopGroup parent and the new one are the same', () => {
+    const groundPlacesService: GroundPlacesController = new GroundPlacesController();
+
+    groundPlacesService.init(largeGroundPlacesFile as GroundPlacesFile);
+
+    const segmentProviderStopId = 'FRBUK';
+    const fromStopGroupGpuid = 'g|FRstraroet@u0tkr3';
+    const intoStopGroupGpuid = 'g|FRstraroet@u0tkr3';
```
Comment (Simobain) [inline __tests__/groundplaces/moveSegmentProviderStop.spec.ts:116]: Well done !

**Review by Theo-Le-Donne — APPROVED** [review]
LGTM ! One question for @Simobain @antonindurieux @rimiti @moussamew : what happens if you move a segmentproviderStop from a StopGoup that only had 1 child ? Notjing happens if he has no children ?

**Comment by Simobain** [comment]
You can have empty stop groups. This is not a problem. @Theo-Le-Donne  👍

**Comment by Theo-Le-Donne** [comment]
> You can have empty stop groups. This is not a problem. @Theo-Le-Donne 👍

Ok perfect ! Was just wondering !

**Comment by Simobain** [comment]
LGTM

**Pull request merged**